v0.7/speech_recognition/rnnt/pytorch/README.md

```diff
@@ -0,0 +1,64 @@
+# 1. Problem 
+Speech recognition accepts raw audio samples and produces a corresponding
+character transcription, without an external language model.
+
+# 2. Directions
+Run `steps/run.sh`.
+
+NOTE: This part will be elaborated on later.
+
+# 3. Dataset/Environment
+### Publication/Attribution
+["OpenSLR LibriSpeech Corpus"](http://www.openslr.org/12/) provides over 1000 hours of speech data in the form of raw audio.
+We use dev-clean, which is approximately 5 hours. (Note: May subsample if this makes the benchmark too slow.)
+### Data preprocessing
+What preprocessing is done to the the dataset?
+
+Log filterbanks of size 80 are extracted every 10 milliseconds, from
+windows of size 20 milliseconds. Note that every three filterbanks are
+concatenated together ("feature splicing"), so the model's effective
+frame rate is actually 30 milliseconds.
+
+TODO: Figure out if preprocessing is considered part of the
```

Comment:
@guschmue Is featurization considered preprocessing? That wasn't clear to me from reading the MLPerf inference paper.

Regardless, mel spectra featurization involves the FFT and logarithm. I wouldn't be surprised if many accelerators lack those primitives. Doing inference directly on features will likely increase submissions in the immediate term.

For perspective, Kaldi (the speech recognition system I worked on for a long time) would normally featurize waveforms to mel spectra on CPU, write to disk, and then run model on top of those. It's fine for offline workloads.

Reply:
Ah, we have history on this. For v0.5 all pre-processing is not timed. But we would have liked to time those as well if it matters for the user. In v0.5 it was out of scope, just didn't have enough time.
For a new model like rnnt - if you think it is crucial to capture those times I'd make that recommendation to the workgroup.

Reply:
This would be a good one for discussion in the WG. MLPerf is a system benchmark rather than an accelerator benchmark, and constructing the Mel Spectra seems like an intrinsic part of the workload.

Are there any online applications where you would expect to get the mel spectra as input?

Reply:
No, there is no scenario in which you would get the mel spectra input, online or not. Ultimately, the input is just a waveform in all cases.

Reply:
To confirm, my belief now is that it makes sense to include MFCC preprocessing as part of the benchmark.